core/include/cubos/core/geom/intersections.hpp

```diff
@@ -0,0 +1,39 @@
+/// @file
+/// @brief Class @ref cubos::core::geom::Intersections.
+/// @ingroup core-geom
+
+#pragma once
+
+#include <vector>
+
+#include <glm/glm.hpp>
+
+#include <cubos/core/geom/box.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+
+namespace cubos::core::geom
+{
+    /// @brief Contains info regarding a collision.
+    /// @ingroup core-geom
+    struct CollisionInfo
+    {
+        CUBOS_REFLECT;
+
+        bool collided;      ///< If there was a collision
+        float penetration;  ///< Penetration depth of the collision.
+        glm::vec3 position; ///< Position of contact on the surface of the entity.
+        glm::vec3 normal;   ///< Normal of contact on the surface of the entity.
+    };
+
+    CollisionInfo BoxBoxIntersection(const Box& box1, const glm::mat4& localToWorld1, const Box& box2,
```

Comment:
```suggestion
    /// docs here...
    CollisionInfo intersects(const Box& box1, const glm::mat4& localToWorld1, const Box& box2,
```

No need to add BoxBox to the name: C++ allows us to have multiple functions with the same name as long as they receive different argument types.

Reply:
Also, I think it would be best to receive a `CollisionInfo&` by reference, where data is written to if there is a collision. Then you could return a boolean in the function, indicatign if there was indeed a collision. Since the struct may get quite big, we avoid unnecessary moving around of data this way.